src/hotspot/cpu/aarch64/register_aarch64.hpp

```diff
@@ -32,62 +32,87 @@
 class VMRegImpl;
 typedef VMRegImpl* VMReg;
 
-// Use Register as shortcut
-class RegisterImpl;
-typedef const RegisterImpl* Register;
+class Register {
+ private:
+  int _encoding;
 
-inline constexpr Register as_Register(int encoding);
+  constexpr Register(int encoding, bool unused) : _encoding(encoding) {}
```

Comment:
Why do we need unused here? If the intention is to prevent implicit conversion from `int` to `Register` then you can mark the constructor as `explicit`. Thanks.

Reply:
Good point. Fixed.